tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsWriter.cs

```diff
@@ -182,6 +185,44 @@ private async Task FlushAndCloseAsync()
         }
     }
 
+    public async Task FlushAsync()
+    {
+        await _flushSemaphore.WaitAsync().ConfigureAwait(false);
+        try
+        {
+            var timeout = TimeSpan.FromSeconds(5);
+
+            if (_processExit.Task.IsCompleted)
+            {
+                return;
+            }
+
+            if (!Volatile.Read(ref _isInitialized) || _processTask == null)
+            {
+                return;
+            }
+
+            var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+            _currentFlushTcs = tcs;
+
+            RequestFlush();
+
+            var completedTask = await Task.WhenAny(
+                tcs.Task,
+                Task.Delay(timeout)).ConfigureAwait(false);
```

Comment:
I think you should maybe also break on process exit? 🤔 
```suggestion
            var completedTask = await Task.WhenAny(
                tcs.Task,
                _processExit.Task,
                Task.Delay(timeout)).ConfigureAwait(false);
```

Reply:
Good call, added